cli/src/test-utils/test-runner.ts

```diff
@@ -13,472 +13,472 @@ import {
-  baseUrl: string,
-  testFilter?: TestFilter
-): Promise<boolean> {
-  const testSuiteStartTime = TestTimer.startTime();
-
-  let allPassed = true;
-
-  for (const endpoint of definition.endpoints) {
-    for (const test of endpoint.tests) {
-      if (testFilter) {
-        if (
-          testFilter.endpoint !== endpoint.name ||
-          (testFilter.test && testFilter.test !== test.name)
-        ) {
-          TestLogger.warn(`Test ${endpoint.name}:${test.name} skipped`);
-          continue;
-        }
-      }
-      const testStartTime = TestTimer.startTime();
+export class TestRunner {
+  private readonly config: TestRunnerConfig;
+  private readonly logger: TestLogger;
 
-      TestLogger.log(`Testing ${endpoint.name}:${test.name}`);
-      const correlatedResponse = findCorrelatedResponse(endpoint, test);
-      const result = await executeTest(
-        test,
-        baseStateUrl,
-        baseUrl,
-        endpoint,
-        correlatedResponse,
-        definition.types
-      );
+  constructor(config: TestRunnerConfig) {
+    this.config = config;
+    this.logger = new TestLogger({ debugMode: config.debugMode });
+  }
 
-      if (result) {
-        TestLogger.success(
-          `Test ${endpoint.name}:${test.name} passed (${TestTimer.formattedDiff(
-            testStartTime
-          )})`,
-          { indent: 1 }
-        );
-      } else {
-        TestLogger.error(
-          `Test ${endpoint.name}:${test.name} failed (${TestTimer.formattedDiff(
-            testStartTime
-          )})`,
-          { indent: 1 }
+  /**
+   * Run the contract test suite for a contract.
+   *
+   * @param definition contract definition
+   * @param config test configuration
+   */
+  async test(
+    definition: ContractDefinition,
+    config?: TestConfig
+  ): Promise<boolean> {
+    const testSuiteStartTime = TestTimer.startTime();
+
+    let allPassed = true;
+
+    for (const endpoint of definition.endpoints) {
+      for (const test of endpoint.tests) {
+        if (config && config.testFilter) {
+          if (
+            config.testFilter.endpoint !== endpoint.name ||
+            (config.testFilter.test && config.testFilter.test !== test.name)
+          ) {
+            this.logger.warn(`Test ${endpoint.name}:${test.name} skipped`);
+            continue;
+          }
+        }
+        const testStartTime = TestTimer.startTime();
+
+        this.logger.log(`Testing ${endpoint.name}:${test.name}`);
+        const correlatedResponse = this.findCorrelatedResponse(endpoint, test);
+        const result = await this.executeTest(
+          test,
+          endpoint,
+          correlatedResponse,
+          definition.types
         );
+
+        if (result) {
+          this.logger.success(
+            `Test ${endpoint.name}:${
+              test.name
+            } passed (${TestTimer.formattedDiff(testStartTime)})`,
+            { indent: 1 }
+          );
+        } else {
+          this.logger.error(
```

Comment:
This does not seem to log the error cause from `executeTest`?

Reply:
The reason for failure is reported during the test run, `result` is simply the boolean to let the test runner know that this test failed - to print out "this test failed" message